cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/preview/PreviewHttpHandler.java

```diff
@@ -70,4 +70,14 @@ public void getPreviewStatus(HttpRequest request, HttpResponder responder,
     responder.sendString(HttpResponseStatus.OK,
                          GSON.toJson(previewManager.getStatus(new PreviewId(namespaceId, previewId))));
```

Comment:
Can you actually GSON a `Throwable` (which is inside `PreviewStatus` returned by the `getStatus` call. I don't see we are using any type adapter to do that.